### PR TITLE
migrate.after_extensive_io: Fix ValueError

### PR DESCRIPTION
**Pull request by bgartzi**
The cause of the raised error has been fixed. Also a related unwanted magnitude conversion has been removed.

ID: 2030589

Signed-off-by: bgartzia <bgartzia@redhat.com>

### CODE REVIEW

**Comment by bgartzi** [comment]
python ConfigTest.py --debug --hostname RHEL-9.0.0 --guestname RHEL.9.0.0 --testcase migrate.after_extensive_io..tcp --clone no

Result:
 (1/2) Host_RHEL.m9.u0.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.migrate.after_extensive_io.nfs.tcp.q35: ERROR: Shell command failed: 'service nfs start'    (status: 5,    output: 'Redirecting to /bin/systemctl start nfs.service\nFailed to start nfs.service: Unit nfs.service not found.\n') (65.45 s)
 (2/2) Host_RHEL.m9.u0.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.migrate.after_extensive_io.iscsi.tcp.q35: ERROR: Shell command failed: 'service tgtd start'    (status: 5,    output: 'Redirecting to /bin/systemctl start tgtd.service\nFailed to start tgtd.service: Unit tgtd.service not found.\n') (65.88 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

Both tests fail. The reason is that test dependencies are not installed in the guest properly. Code should be adapted to use nfs and iscsi server implementations from virttest.

**Comment by bgartzi** [comment]
@YongxueHong @luckyh would you review the request and help me through the process? Thank you.

File: qemu/tests/migration_with_dst_problem.py [hunk 1]
```diff
@@ -438,8 +438,8 @@ def test(self):
             self.copy_timeout = int(params.get("copy_timeout", "1024"))
 
             self.copy_block_size = int(utils_misc.normalize_data_size(
-                params.get("copy_block_size", "100M"), "M"))
-            self.disk_size = "%sM" % int(self.copy_block_size * 1.4)
+               params.get("copy_block_size", "100M"), "M").split(".")[0])
+            self.disk_size = int(self.copy_block_size * 1.4)
```
Comment (YongxueHong) [inline qemu/tests/migration_with_dst_problem.py:442]: From your commit message header: `Fix ValueError: invalid literal for int() with base 10`, I think we can only focus on it.
And for line 442, 455 and 631~632, I guess it maybe another issue about the design flaw.

**Comment by bgartzi** [comment]
Suggested changes have been applied.
Now the result of running 
` python ConfigTest.py --debug --hostname RHEL-9.0.0 --guestname RHEL.9.0.0 --testcase migrate.after_extensive_io..tcp --clone no ` is
(1/2) Host_RHEL.m9.u0.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.migrate.after_extensive_io.nfs.tcp.q35: ERROR: Shell command failed: 'service nfs start'    (status: 5,    output: 'Redirecting to /bin/systemctl start nfs.service\nFailed to start nfs.service: Unit nfs.service not found.\n') (64.44 s)
 (2/2) Host_RHEL.m9.u0.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.migrate.after_extensive_io.iscsi.tcp.q35: ERROR: could not convert string to float: '140M' (56.43 s)

**Review by YongxueHong — APPROVED** [review]

**Comment by YongxueHong** [comment]
Hi, @leexiaohui32
Can you help to review it?
Thanks.

**Comment by leexiaohui32** [comment]
Ack on my side.

**Pull request merged**